### DISCUSSION TITLE
间歇性无法正常连接

### DISCUSSION THREAD

**Discussion by wning13** [v2fly/v2ray-core#2403] [Q&A]
我使用的是vmess + ws + tls + Cloudflare CDN的方案，在大多数情况下可以正常工作。但有时会出现某个客户端无法正常连接的情况。通常情况下，其他IP的客户端仍然可以正常使用。这种情况通常会持续一段时间，然后恢复正常。
出错时客户端的日志如下：
app/proxyman/outbound: failed to process outbound traffic > proxy/vmess/outbound: failed to find an available destination > common/retry: [transport/internet/websocket: failed to dial WebSocket > transport/internet/websocket: failed to dial to (wss://XXXX/ray):  > dial tcp XXXX:443: i/o timeout transport/internet/websocket: failed to dial WebSocket > transport/internet/websocket: failed to dial to (wss://XXXXX/ray):  > dial tcp: operation was canceled] > common/retry: all retry attempts failed
VPS上nginx和v2ray的日志都没有相应请求的记录。
请问这种情况是什么原因造成的？有什么方法可以减少或规避这样的问题？

**Comment by mydogshitgold** [comment]
The edge IP you are using is probably being blocked. Check your tcpdump results to see if there are any abnormal RST packets.

You can manually switch to another IP (all Cloudflare edge IPs are anycast). Check out this [BGP dump](https://bgp.tools/table.jsonl) to find all Cloudflare prefixes that have an AS number of 13335, 209242, 202623, 395747, 139242, 203898, 132892, 394536 or 209242. Not all prefixes are anycast, you will need to run automated tests in an unrestricted environment (or your ISP may have cut your cable for "DoS") to find them out. There are [a number](https://github.com/XIU2/CloudflareSpeedTest) of [projects](https://github.com/badafans/better-cloudflare-ip) on GitHub that do this. You can also find out by running `curl <ip>` and seeing if the output is `error code: 1003`.

If changing the IP doesn't help, it's likely that your domain name is blocked. Then God bless you.

Or you can just wait for a while.

---

> What is the reason for this situation? Is there any way to reduce or circumvent such problems?

Better [camouflage](https://guide.v2fly.org/advanced/wss_and_web.html), superior [anonymity](https://anonymousplanet-ng.org/guide.html), stronger [plausible deniability](https://github.com/v2fly/v2ray-core/pull/2219) and [luck](https://i.kym-cdn.com/entries/icons/original/000/035/146/knight.jpg).

---

NOTE: Most traditional blocking is implemented as 5 tuple (src ip, dest ip, src port, dest port, protocol), switch to another source IP or ISP may help.

**Reply by mydogshitgold** [reply]
Here is an example of using a custom anycast IP with reverse proxied tunnel
```json
{
  "outbounds": [
    {
      "protocol": "vmess",
      "settings": {
        "vnext": [
          {
            "address": "104.16.132.229",
            "port": 443,
            "users": [
              {
                "id": "00000000-0000-0000-0000-000000000000"
              }
            ]
          }
        ]
      },
      "streamSettings": {
        "network": "ws",
        "security": "tls",
        "wsSettings": {
          "path": "/tunnel?s=vf-ovh-inst21&u=testing",
          "headers": {
            "Host": "my.asshole.ovh"
          }
        },
        "tlsSettings": {
          "serverName": "my.asshole.ovh"
        }
      }
    }
  ]
}
```

**Reply by wning13** [reply]
thanks，it works for me

**Comment by nursery01** [comment]
> 这种情况是什么原因造成的？有什么方法可以减少或规避这样的问题？

可以試一試用中轉伺服器作爲跳板連結v2ray，使用分散式系統
定期更換ip和埠，使用多種協議

```
client1------------|----------intermediate server1----------------|
                   |                         ......               |------------v2ray17
......             |----------intermediate server32---------------|            ......
                  |                       ......
client99----------|-----------intermediate server54---------------|------------v2ray42
```
中轉伺服器可以定期發送接收隨機數據，僞裝自己是有正常業務的伺服器

**Comment by 1265578519** [comment]
客户端上的 tls设置 允许不安全就可以了
完美解决办法 去掉HTTPS就不会被干扰了
